File: src/main/java/com/tribalscale/test/domain/payload/AccountPayload.java
package com.tribalscale.test.domain.payload;

import io.swagger.annotations.ApiModelProperty;
import lombok.Builder;
import lombok.Value;
import lombok.extern.jackson.Jacksonized;

import javax.validation.constraints.NotBlank;
import javax.validation.constraints.Pattern;

@Value
@Builder
@Jacksonized
public class AccountPayload {
    @ApiModelProperty(notes = "The account id (alphanumeric)", name = "accountId", required = true, value = "11aa23")
    @NotBlank
    @Pattern(regexp = "^[a-zA-Z0-9\\s]*$")
    String accountId;

    @ApiModelProperty(notes = "The ISO code for the currency", name = "currency", required = true, value = "EUR")
    @NotBlank
    @Pattern(regexp = "^[a-zA-Z\\s]*$")
    String currency;

    @ApiModelProperty(notes = "The final balance for this account", name = "balance", required = true, value = "1000.0")
    Double balance;
}

File: src/main/java/com/tribalscale/test/domain/payload/AccountPayload.kt
package com.tribalscale.test.domain.payload

import io.swagger.v3.oas.annotations.media.Schema
import lombok.Builder
import lombok.Value
import lombok.extern.jackson.Jacksonized
import javax.validation.constraints.NotBlank
import javax.validation.constraints.Pattern

@Value
@Builder
@Jacksonized
class AccountPayload(
    @Schema(description = "The account id (alphanumeric)", name = "accountId", required = true, example = "11aa23")
    val accountId: @NotBlank @Pattern(regexp = "^[a-zA-Z\\d\\s]*$") String,

    @Schema(description = "The ISO code for the currency", name = "currency", required = true, example = "EUR")
    val currency: @NotBlank @Pattern(regexp = "^[a-zA-Z\\s]*$") String,

    @Schema(description = "The final balance for this account", name = "balance", required = true, example = "1000.0")
    val balance: Double
)